zerver/views/webhooks/codeship.py

```diff
@@ -0,0 +1,47 @@
+# Webhooks for external integrations.
+from __future__ import absolute_import
+from zerver.models import get_client
+from zerver.lib.actions import check_send_message
+from zerver.lib.response import json_success
+from zerver.decorator import REQ, has_request_variables, api_key_only_webhook_view
+
+import ujson
+
+
+CODESHIP_SUBJECT_TEMPLATE = '{project_name} build status.'
+CODESHIP_MESSAGE_TEMPLATE = '[Build]({build_url}) triggered by {committer} on {branch} branch {status}.'
+
+CODESHIP_DEFAULT_STATUS = 'has {status} status'
+CODESHIP_STATUS_MAPPER = {
+    'testing': 'started',
+    'error': 'failed',
+    'success': 'succeeded',
+}
+
+
+@api_key_only_webhook_view
+@has_request_variables
+def api_codeship_webhook(request, user_profile, stream=REQ(default='codeship')):
+    payload = ujson.loads(request.body)['build']
+    subject = get_subject_for_http_request(payload)
+    body = get_body_for_http_request(payload)
+    check_send_message(user_profile, get_client('ZulipCodeshipWebhook'), 'stream', [stream], subject, body)
+    return json_success()
```

Comment:
Can you add some error-handling in the parsing similar to what we have in zerver/views/webhooks/stash.py ?